### ISSUE TITLE
New feature request:implementation of _get_control_flow_context function

### ISSUE THREAD

**Issue by sepidehsaran** [tensorflow/tensorflow#46402]
For interpretability of siamese-like networks with two legs (blocks of BiLSTM), I need to get the gradients of output (similarity value between 0-1) with respect to two inputs (two sequence) for which I use the gradient tape. But then I run to the following error: 

_tf.GradientTape.gradients() does not support graph control flow operations like tf.cond or tf.while at this time. Use tf.gradients() instead. If you need this feature, please file a feature request at https://github.com/tensorflow/tensorflow/issues/new_

coming from: tensor flow/python/eager/backprop.py - def _get_control_flow_context

Could you please explain how this can be solved? Can use of this function be avoided for now or can you please explain what I need to implement there?
 
Training the network was fine and prediction results look good. The issue is only with interpretability of the method. Here is where my code faces this error:

```
# convert to tensor and reshape
sample_seq_1 = tf.convert_to_tensor(train_seq1[0], dtype=tf.float32)
sample_seq_2 = tf.convert_to_tensor(train_seq2[0], dtype=tf.float32)
sample_seq_1 = tf.expand_dims(sample_seq_1, 0)
sample_seq_2 = tf.expand_dims(sample_seq_2, 0)

# inference in tape
with tf.GradientTape(persistent=True) as tape:
    tape.watch([sample_seq_1, sample_seq_2])
    similarity_value_pred = siamese_net_model([sample_seq_1, sample_seq_2])[0,0]

# getting gradients
tape.gradient(similarity_value_pred, [sample_seq_1, sample_seq_1])
```

**Comment by ravikyram** [comment]
@sepidehsaran 

Please, fill [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).
Please, share colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by sepidehsaran** [comment]
I have created [this notebook](https://colab.research.google.com/drive/1k-FCSkw6CfGyMkwR5eTS4U8-jGVzd24I?usp=sharing) to reproduce the issue on dummy data. 
I have realised the notImplemented error only occurs when using TF 1.x. Using TF 2.4 I get none for the gradients.
I could appreciate it if you could suggest a solution that would work with either 1.x or 2.x. Thanks!

**Comment by tilakrayal** [comment]
I was able to reproduce the issue on tf-nightly 2.10.0-dev20220720. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/118748b7d2fa6e889792c98172c7e689/nightlyreproducing_tf_issue_46402-1.ipynb). Thank you!

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
